intact-orthology-import/src/main/java/uk/ac/ebi/intact/ortholog/OrthologsXrefWriter.java

```diff
@@ -0,0 +1,51 @@
+package uk.ac.ebi.intact.ortholog;
+
+import lombok.RequiredArgsConstructor;
+import psidev.psi.mi.jami.model.Xref;
+import uk.ac.ebi.intact.jami.dao.IntactDao;
+import uk.ac.ebi.intact.jami.model.extension.IntactCvTerm;
+import uk.ac.ebi.intact.jami.model.extension.IntactProtein;
+import uk.ac.ebi.intact.jami.model.extension.InteractorXref;
+import uk.ac.ebi.intact.jami.utils.IntactUtils;
+import java.util.*;
+import java.util.Map;
+
+@RequiredArgsConstructor
+public class OrthologsXrefWriter {
+
+    private final static String PANTHER_DATABASE_MI = "MI:0702";
+    private final Map<String, IntactCvTerm> cvTermMap = new HashMap<>();
+    private final IntactDao intactDao;
+
+    public void addOrthologyXrefs(IntactProtein protein, Collection<String> pantherIds) throws Exception{
+        for (String pantherId: pantherIds) {
+            addOrthologyXref(protein, pantherId);
+        }
+    }
+
+    public void addOrthologyXref(IntactProtein protein, String pantherId) throws Exception{
+        InteractorXref xref = newOrthologsXref(pantherId);
+        if (!protein.getXrefs().contains(xref)){
```

Comment:
If this if doesn't work, then we can do:
```
        Collection<Xref> existingOrthologXrefs = protein.getXrefs()
                .stream()
                .filter(id -> PANTHER_DATABASE_MI.equals(id.getDatabase().getMIIdentifier()))
                .filter(id -> Xref.SEE_ALSO_MI.equals(id.getQualifier().getMIIdentifier()))
                .collect(Collectors.toList());

        if (existingOrthologXrefs.stream().noneMatch(xref -> xref.getId().equals(pantherId))) {
            InteractorXref xref = newOrthologsXref(pantherId);
            if (!protein.getXrefs().contains(xref)) {
                protein.getXrefs().add(xref);
            }
        }
```

We get all xrefs from the protein with the panther db and see-also qualifier, and we check if any matches with the panther id we are trying to save. If any does, then it already exists.

At the moment we don't need to worry about it, but for future runs we need to think about what we want to do when we run this import a second time. It's similar to the uniplex import. Options are:
1. Delete all panther xrefs and start an import from clean
2. Update the code to compare protein-panther mappings with the ones in the DB. If they match, all good; if they don't, we update it. We would also need to check any protein with a panther id saved, which no longer maps to it, which we would need to update to.

As said, not to worry about this now.